src/libraries/System.Text.Json/gen/Reflection/TypeWrapper.cs

```diff
@@ -382,6 +388,12 @@ public override PropertyInfo[] GetProperties(BindingFlags bindingAttr)
             {
                 if (item is IPropertySymbol propertySymbol)
                 {
+                    // Skip auto-generated properties on records.
```

Comment:
How do we do this for the reflection side?

Reply:
I looked briefly through the reflection implementation but didn't find anything special-casing record types. Without this filtration here, we end up generating code for a `System.Assembly`-typed property that was implicitly generated (which causes problems in user code due to the property being deprecated).

For now the goal with these Roslyn-Reflection APIs is to get the serializer tests to all pass. At a time when we want to ship these helpers, we'd subject them to the full set of System.Reflection tests to ensure behavioral parity.